src/price-feeds/chainlink/LibChainlinkPriceFeed.sol

```diff
@@ -0,0 +1,187 @@
+    internal
+    view
+    returns (uint256 tokenInAmount)
+  {
+    (uint256 price, uint8 priceDecimal) = quotePrice(priceFeed);
+
+    // Scale the price to the same decimal as tokenIn
+    // The price is in tokenOut, so we need to inverseAndScalePrice it to get the tokenIn price
+    uint256 inversedPrice = inverseAndScalePrice(price, priceDecimal, priceFeed._tokenInDecimal);
+
+    tokenInAmount = Math.mulDiv(inversedPrice, tokenOutAmount, 10 ** priceFeed._tokenOutDecimal);
+  }
+
+  /**
+   * @dev Get the price from the Chainlink price feed.
+   * @param priceFeed The Chainlink price feed.
+   * @return price The price in the given decimal.
+   */
+  function quotePrice(ChainlinkPriceFeed memory priceFeed) internal view returns (uint256 price, uint8 priceDecimal) {
+    (, int256 answer,, uint256 updatedAt,) = priceFeed._aggregator.latestRoundData();
+    if (updatedAt < block.timestamp - priceFeed._maxAcceptableAge) {
+      revert PriceTooOld(updatedAt, block.timestamp - priceFeed._maxAcceptableAge);
+    }
+    if (answer <= 0) revert PanicNegativeQuotePrice(answer);
+
+    priceDecimal = priceFeed._aggregator.decimals();
+    if (priceDecimal > _DECIMAL_LIMIT) revert LargeDecimal(priceDecimal);
+
+    return (uint256(answer), priceDecimal);
+  }
+
+  /**
+   * @dev Scale the price to the given decimal.
+   * @param price The price in the given decimal.
+   * @param priceDecimal The decimal of the price.
+   * @param scaleDecimal The decimal to scale the price to.
+   * @return scaledPrice The scaled price in the given decimal.
+   */
+  function scalePrice(uint256 price, uint8 priceDecimal, uint8 scaleDecimal)
+    internal
+    pure
+    returns (uint256 scaledPrice)
+  {
+    uint256 log10Price = Math.log10(price);
+    if (scaleDecimal > priceDecimal && log10Price + (scaleDecimal - priceDecimal) > _MAX_DECIMAL) {
+      revert ComputedPriceTooLarge(price, -(int8(scaleDecimal) - int8(priceDecimal)));
+    }
+    if (scaleDecimal < priceDecimal && log10Price < priceDecimal - scaleDecimal) {
+      revert ComputedPriceTooSmall(price, -(int8(priceDecimal) - int8(scaleDecimal)));
+    }
+
+    return price.exp10(int8(scaleDecimal) - int8(priceDecimal));
+  }
+
+  /**
+   * @dev Inverse the price of (A/B) to (B/A) and scale it to the given decimal.
+   * @param price The price of (A/B) in the given decimal.
+   * @param priceDecimal The decimal of the price.
+   * @param scaleDecimal The decimal to scale the price to.
+   * @return inversedPrice The price of (B/A) scaled in the given decimal.
+   */
+  function inverseAndScalePrice(uint256 price, uint8 priceDecimal, uint8 scaleDecimal)
+    internal
+    pure
+    returns (uint256 inversedPrice)
+  {
+    if (Math.log10(price) > priceDecimal + scaleDecimal) {
+      revert ComputedPriceTooSmall(price, -(int8(scaleDecimal) - int8(priceDecimal)));
+    }
+
+    return 10 ** (scaleDecimal + priceDecimal) / price;
```

Comment:
If I understand correctly, scaling the price means `(current price / 10 ^ price decimal) * 10 ^ desired decimal`. This calculation is another way of the formula `(1 * 10 ^ scaled decimal) / (p * 10 ^ (-price decimal))`. If this is correct, please add a comment to enhance readability.

Also, add a comment explaining why we don’t re-invoke the `scalePrice` function for `1/price`. I assume it's to avoid rounding to zero, right?